### PR TITLE
Add actions and Selenium Grid

### PR DESCRIPTION
**Pull request by ArinaNichko**
Tests passed - https://www.dropbox.com/s/tny9ck4oumh6eoq/photo_2023-03-17_16-42-40.jpg?dl=0

### CODE REVIEW

File: src/test/java/pages/HomePage.java [hunk 1]
```diff
@@ -4,42 +4,48 @@
...
   @FindBy(id = "dropdown-checkboxes-radiobuttons")
-  private WebElement dropdownLinks;
+  private WebElement optionsLink;
 
   @FindBy(css = "#popup-alerts")
   private WebElement popupAndAlertsLink;
 
-  @FindBy(xpath = "//*[@id=\"button1\"]")
-  private WebElement popupButton;
+  @FindBy(xpath = "//h1[contains(text(),'ACCORDION')]")
```
Comment (ivanKuleshin) [inline src/test/java/pages/HomePage.java:16]: Why did you change xpath from searching by id to searching by text?

**Comment by ivanKuleshin** [comment]
BaseTest class should be located with all test cases, it's not a util class. For example:
- testcases
    - BaseTest.class
    - featurefolder
       - featureTest.java

File: src/test/java/testCases/AccordionItemsTests.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  @Test
+  public void checkKeepClickingItem() {
+    homePage.openHomePage();
+    homePage.clickOnAccordionItemsLink();
+
+    List<String> tabsList = homePage.tabsList();
+    homePage.switchToTab(tabsList.get(FIRST));
+
+    accordionItemsPage.implicitWait(TIMEOUT);
```
Comment (ivanKuleshin) [inline src/test/java/testCases/AccordionItemsTests.java]: Why do you need implicitWait here?

File: src/test/java/utils/BaseTest.java [hunk 3]
```diff
@@ -1,39 +1,71 @@
...
+    setUpDriver(browser);
     driver.manage().window().maximize();
     pageFactoryManager = new PageFactoryManager(driver);
-    homePage = pageFactoryManager.getHomePage();
-    alertAndPopupPage = pageFactoryManager.getAlertAndPopupPage();
+    initPages();
+  }
 
+  public void setUpDriver(String browser) {
+    if (browser.equalsIgnoreCase("chrome")) {
```
Comment (ivanKuleshin) [inline src/test/java/utils/BaseTest.java]: It's better to move this logic outside this method. For example, into the method getDesiredCapabilities. This will improve readability

File: src/test/java/utils/BaseTest.java [hunk 4]
```diff
@@ -1,39 +1,71 @@
...
+      capabilities = DesiredCapabilities.chrome();
+    } else if (browser.equalsIgnoreCase("firefox")) {
+      capabilities = DesiredCapabilities.firefox();
+    } else if (browser.equalsIgnoreCase("MicrosoftEdge")) {
```
Comment (ivanKuleshin) [inline src/test/java/utils/BaseTest.java]: Standardize browser names

File: src/test/java/utils/BaseTest.java [hunk 5]
```diff
@@ -1,39 +1,71 @@
...
+    }
+    try {
+      driver = new RemoteWebDriver(new URL("http://172.20.10.3:4444/"), capabilities);
+    } catch (MalformedURLException e) {
+      throw new TestExecutionException(e.getMessage());
+    }
   }
 
-  @AfterClass
+  public void initPages() {
```
Comment (ivanKuleshin) [inline src/test/java/utils/BaseTest.java]: Are you sure, that all pages require all other pages to be extended?
For example, optionsPage variable is used only in the OptionsTests class. So, LoginPortalTests class and all others do not require it. But they anyway have unnecessary access to optionsPage variable, because it is marked as protected in the BaseTest...

File: src/test/java/utils/BaseTest.java [hunk 6]
```diff
@@ -1,39 +1,71 @@
...
+    homePage = pageFactoryManager.getHomePage();
```
Comment (ivanKuleshin) [inline src/test/java/utils/BaseTest.java]: I think, that only homePage should be shared between test cases.

File: src/test/java/utils/BaseTest.java [hunk 7]
```diff
@@ -1,39 +1,71 @@
...
+  public void setUpDriver(String browser) {
+    if (browser.equalsIgnoreCase("chrome")) {
+      capabilities = DesiredCapabilities.chrome();
+    } else if (browser.equalsIgnoreCase("firefox")) {
+      capabilities = DesiredCapabilities.firefox();
+    } else if (browser.equalsIgnoreCase("MicrosoftEdge")) {
+      capabilities = DesiredCapabilities.edge();
+    }
+    try {
+      driver = new RemoteWebDriver(new URL("http://172.20.10.3:4444/"), capabilities);
```
Comment (ivanKuleshin) [inline src/test/java/utils/BaseTest.java]: Please add the ability to run tests locally. For example, depending on the boolean localRun parameter you can implement remote or local Webdriver. This parameter can be taken from a general test suite runner or from a specific XML runner.

**Review by ivanKuleshin — CHANGES_REQUESTED** [review]

File: src/test/java/pages/HomePage.java (hunk 1 above)
Comment (ArinaNichko) [reply src/test/java/pages/HomePage.java:16]: Because I searched by id using last()

**Comment by ivanKuleshin** [comment]
small improvements: 
- common classes, like BasePage or BaseTest should be clearly visible;
- test classes should be in separate folders, according to business components
- name convention for all packages should be met

**Review by ivanKuleshin — APPROVED** [review]
